lib/torrent.js

```diff
@@ -1029,36 +1035,37 @@ export default class Torrent extends EventEmitter {
 
     this._debug('select %s-%s (priority %s)', start, end, priority)
 
-    this._selections.push({
+    this._selections.insert({
       from: start,
       to: end,
       offset: 0,
       priority,
       notify: notify || noop
-    })
+    }, isStreamSelection)
 
-    this._selections.sort((a, b) => b.priority - a.priority)
+    this._selections.sort((a, b) => a.priority - b.priority)
```

Comment:
why was the order reversed?

Reply:
honestly no clue. probably just an accident, but good catch. I reverted to the old order